### PR TITLE
chore: Add `--throttle` flag to `propagate-reporting` command

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

**Comment by alfrunes** [comment]
@mender-test-bot start pipeline --fast

File: main.go [hunk 1]
```diff
@@ -91,6 +91,11 @@ func doMain(args []string) {
 					Name:  "tenant_id",
 					Usage: "Tenant ID (optional) - propagate for just a single tenant.",
 				},
+				cli.DurationFlag{
+					Name:  "throttle",
+					Usage: "`DURATION` to wait for each reindexing batch request",
```
Comment (merlin-northern) [inline main.go:96]: please sespecifycify what is a DURATION with units.

Comment (merlin-northern) [reply main.go:96]: also I find is slightly confusing name of the option is throttle and the usage says duration.

File: main.go [hunk 2]
```diff
@@ -91,6 +91,11 @@ func doMain(args []string) {
...
+					Value: time.Millisecond * 25,
```
Comment (merlin-northern) [inline main.go:97]: could we have a const out of it? good news is: you can use it in addressing comments above :)

File: main.go (hunk 1 above)
Comment (merlin-northern) [reply main.go:96]: I have not checked but will we have some examples of the command in the help? and: do we need to update the docs.mender.io?

File: propagete_reporting_test.go [hunk 3]
```diff
@@ -131,7 +132,7 @@ func TestPropagateReporting(t *testing.T) {
 		t.Run(fmt.Sprintf("tc %s", k), func(t *testing.T) {
 			defer tc.workflowsMock.AssertExpectations(t)
 			defer tc.storeMock.AssertExpectations(t)
-			err := propagateReporting(tc.storeMock, tc.workflowsMock, tc.cmdTenant, tc.cmdDryRun)
+			err := propagateReporting(tc.storeMock, tc.workflowsMock, tc.cmdTenant, time.Microsecond, tc.cmdDryRun)
```
Comment (merlin-northern) [inline propagete_reporting_test.go:135]: we could create a new test case, which first calculates the avg time it took to run with throttle eq 0 and then sets the throttle to 2s and verifies that it took on avg longer, and the longer is propotional to the duration.

File: main.go [hunk 4]
```diff
@@ -375,6 +385,8 @@ func reindexDeploymentsReporting(
 		if len(dd) < deviceDeploymentsBatchSize {
 			break
 		}
+		nextIterAt.Add(throttleDuration)
+		time.Sleep(time.Until(nextIterAt))
```
Comment (merlin-northern) [inline main.go:389]: I understand why you are doing it like that, but to be mean: what in case of the time change?

Comment (alfrunes) [reply main.go:389]: Right, I should use a timer to ensure that it's using monotonic time.

File: main.go (hunk 1 above)
Comment (alfrunes) [reply main.go]: Changed the flat to `--rate-limit`. Should be more self-explanatory.

**Review by kjaskiewiczz — APPROVED** [review]

**Pull request merged**